spec/unit/actions/synchronize_processes_spec.rb

```diff
@@ -0,0 +1,110 @@
+                                           from: src_pid,
+                                           to: [
+                                             {
+                                               pid: dest_pid,
+                                               client_id: 'aaaa'
+                                             }
+                                           ]
+                                         ])
+  end
+  let(:process_component) do
+    { name: 'etl-csv-uploader' }
+  end
+  let(:process_hash) do
+    { process: { component: process_component } }
+  end
+  let(:process_additional_hidden_params) do
+    {
+      process: {
+        component: {
+          secretConfig: {
+            key: 'val'
+          }
+        }
+      }
+    }
+  end
+  let(:enriched_process_hash) do
+
+  end
+
+  let(:params) do
+    GoodData::LCM2.convert_to_smart_hash(
+      synchronize: synchronize,
+      development_client: development_client,
+      gdc_gd_client: gdc_gd_client,
+      gdc_logger: gdc_logger,
+      process_additional_hidden_params: process_additional_hidden_params
+    )
+  end
+  before do
+    allow(development_client).to receive(:projects).with(src_pid).and_return(src_project)
+    allow(gdc_gd_client).to receive(:projects).with(dest_pid).and_return(dest_project)
+
+    allow(src_project).to receive(:title).and_return('src title')
+    allow(dest_project).to receive(:title).and_return('dest title')
+    allow(src_project).to receive(:pid).and_return(src_pid)
+    allow(dest_project).to receive(:pid).and_return(dest_pid)
+    allow(src_project).to receive(:processes).and_return([process])
+    allow(dest_project).to receive(:processes).and_return([])
+    allow(dest_project).to receive(:client).and_return(dest_client)
+    allow(src_project).to receive(:add).and_return(add)
+    allow(dest_project).to receive(:add).and_return(add)
+
+    allow(add).to receive(:output_stage).and_return(output_stage)
+    allow(output_stage).to receive(:output_stage_prefix).and_return('prefix')
+    allow(output_stage).to receive(:schema).and_return({})
+    allow(output_stage).to receive(:client_id=)
+
+    allow(gdc_logger).to receive(:info)
+
+    allow(process).to receive(:name).and_return('process name')
+    allow(process).to receive(:type).and_return('etl')
+    allow(process).to receive(:path).and_return(nil)
+    allow(process).to receive(:component).and_return(process_component)
+    allow(process).to receive(:to_hash).and_return(process_hash)
+
+    allow(process).to receive(:project).and_return(src_project)
+  end
+  it 'merges the process params with the component part' do
+    expect(GoodData::Process).to receive(:deploy_component).with(
+      {
+        process: {
+          component: {
+            name: 'etl-csv-uploader',
+            secretConfig: {
+              key: 'val'
+            }
+          }
+        }
+      },
+      {
+        client: dest_client,
+        project: dest_project
+      })
```

Comment:
warning: Layout/MultilineMethodCallBraceLayout: Closing method call brace must be on the line after the last argument when opening brace is on a separate line from the first argument. (rubocop)